src/ragas/config.py

```diff
@@ -1,27 +1,32 @@
+from __future__ import annotations
+
 import typing as t
 
 from pydantic import BaseModel, Field
 
-from ragas.embeddings import BaseRagasEmbeddings
-from ragas.llms import BaseRagasLLM
+from ragas.embeddings import BaseRagasEmbeddings, embedding_factory
+from ragas.llms import BaseRagasLLM, llm_factory
 from ragas.losses import Loss
 from ragas.optimizers import GeneticOptimizer, Optimizer
 
 DEFAULT_OPTIMIZER_CONFIG = {"max_steps": 100}
 
 
 class DemonstrationConfig(BaseModel):
+    embedding: BaseRagasEmbeddings = Field(default_factory=embedding_factory)
     enabled: bool = True
     top_k: int = 3
     technique: t.Literal["random", "similarity"] = "similarity"
-    embedding: t.Optional[BaseRagasEmbeddings] = None
 
 
 class InstructionConfig(BaseModel):
+    llm: BaseRagasLLM = Field(default_factory=llm_factory)
```

Comment:
My though process for making it None was
- If no model is explicitly specified, the optimization should proceed by taking the model instantiated in metric. This will allow seamless optimization with people using custom models other than open_ai.

Reply:
ohh yes that makes much more sense